codec/packer.go

```diff
@@ -17,166 +17,112 @@ import (
 // added to many unpacking methods, which signals the packer to add an error
 // if the expected method does not unpack properly.
 type Packer struct {
-	p *wrappers.Packer
+	*wrappers.Packer
 }
 
 // NewReader returns a Packer instance with the current byte array set to [byte]
 // and it's MaxSize set to [limit].
 func NewReader(src []byte, limit int) *Packer {
 	return &Packer{
-		p: &wrappers.Packer{Bytes: src, MaxSize: limit},
+		Packer: &wrappers.Packer{Bytes: src, MaxSize: limit},
 	}
 }
 
 // NewWriter returns a Packer instance with an initial size of [initial] and a
 // MaxSize set to [limit].
 func NewWriter(initial, limit int) *Packer {
 	return &Packer{
-		p: &wrappers.Packer{Bytes: make([]byte, 0, initial), MaxSize: limit},
+		Packer: &wrappers.Packer{Bytes: make([]byte, 0, initial), MaxSize: limit},
 	}
 }
 
-func (p *Packer) PackBool(src bool) {
-	p.p.PackBool(src)
-}
-
-func (p *Packer) UnpackBool() bool {
-	return p.p.UnpackBool()
-}
-
-func (p *Packer) PackID(src ids.ID) {
-	p.p.PackFixedBytes(src[:])
+func (p *Packer) PackID(id ids.ID) {
+	p.Packer.PackFixedBytes(id[:])
 }
 
 // UnpackID unpacks an avalanchego ID into [dest]. If [required] is true,
 // and the unpacked bytes are empty, Packer will add an ErrFieldNotPopulated error.
 func (p *Packer) UnpackID(required bool, dest *ids.ID) {
-	copy((*dest)[:], p.p.UnpackFixedBytes(ids.IDLen))
+	copy((*dest)[:], p.Packer.UnpackFixedBytes(ids.IDLen))
 	if required && *dest == ids.Empty {
-		p.addErr(fmt.Errorf("%w: ID field is not populated", ErrFieldNotPopulated))
+		p.Packer.Errs.Add(fmt.Errorf("%w: ID field is not populated", ErrFieldNotPopulated))
 	}
 }
 
-func (p *Packer) PackByte(b byte) {
-	p.p.PackByte(b)
-}
-
-func (p *Packer) UnpackByte() byte {
-	return p.p.UnpackByte()
-}
-
-func (p *Packer) PackFixedBytes(b []byte) {
-	p.p.PackFixedBytes(b)
-}
-
+// FIXME: do we need this or UnpackFixedBytes is enough?
```

Comment:
Let's keep this, since `UnpackFixedBytes` is not the same as `UnpackAddress`, I think it makes sense to keep both `PackAddress` and `UnpackAddress` with the extra empty address check, so that it's symmetric.